wrong case for naming / namespace

In the generated code below, you can see the `Message` fragment is referred as `${Message.Document}`, but it should have been `${message.Document}`. 

```ts

export namespace Message {
  export type Fragment = {
    __typename?: 'Message';
    id: string;
    text?: string | null;
    status?: string | null;
    createdAt: Time;
  };
}

export namespace message {
  export const Document = gql`
    fragment message on Message {
      id
      text
      status
      createdAt
    }
  `;
}

export namespace CreateMessage {
  export const Document = gql`
    mutation createMessage($input: CreateMessageInput!) {
      createMessage(input: $input) {
        message {
          ...message
        }
      }
    }

    ${Message.Document}
  `;
  export class Component extends React.Component<
    Partial<ReactApollo.MutationProps<Mutation, Variables>>
  > {
    render() {
      return (
        <ReactApollo.Mutation<Mutation, Variables>
          mutation={Document}
          {...this.props as any}
        />
      );
    }
  }
  export function HOC<TProps = {}>(
    operationOptions?: ReactApollo.OperationOption<TProps, Mutation, Variables>
  ) {
    return ReactApollo.graphql<TProps, Mutation, Variables>(
      Document,
      operationOptions
    );
  }
}
```

my fragment is defined as 

```
export const messageFragment = gql`
  fragment message on Message {
    id
    text
    status
    createdAt
  }
`;

```

I think a better name for namespace would be `messageFragment`. Is it easily configurable?

thanks.

that solved the issue with case sensitivity, but there is two namespace with same name. Hope that is ok.

```
export namespace Message {
	export const Document = gql`
		fragment message on Message {

....

```

```
export namespace Message {
	export type Fragment = {
		__typename?: "Message";
..
```

But, I have bunch of errors (which was present earlier with stable release as well)

```

src/graphql/types.ts:1173:11 - error TS1005: '>' expected.

1173           subscription={Document}
               ~~~~~~~~~~~~

src/graphql/types.ts:1173:23 - error TS1005: ')' expected.

1173           subscription={Document}
                           ~

src/graphql/types.ts:1174:12 - error TS1128: Declaration or statement expected.

1174           {...this['props'] as any}
                ~~~

src/graphql/types.ts:1175:10 - error TS1161: Unterminated regular expression literal.

1175         />


src/graphql/types.ts:1176:7 - error TS1128: Declaration or statement expected.

1176       );
           ~
```

and the offending code is

```
1171 .  return (
1172 .        <ReactApollo.Subscription<Subscription, Variables>
1173 .          subscription={Document}
1174 .          {...this['props'] as any}
 1175 .       />
1176 .      );

```

do I need to change my tsconfig?

--- a/packages/templates/apollo-angular/tests/components.spec.ts
+++ b/packages/templates/apollo-angular/tests/components.spec.ts
@@ -13,15 +13,6 @@ import config from '../dist';
 import * as fs from 'fs';
 import { DocumentNode, extendSchema } from 'graphql';
 
-const compileAndBuildContext = (typeDefs: string): { context: SchemaTemplateContext; schema: GraphQLSchema } => {
-  const schema = makeExecutableSchema({ typeDefs, resolvers: {}, allowUndefinedInResolve: true });
-
-  return {
-    schema,
-    context: schemaToTemplateContext(schema)
-  };
-};
-
 describe('Components', () => {
   it('should generate Component', async () => {
     const schema = introspectionToGraphQLSchema(JSON.parse(fs.readFileSync('./tests/files/schema.json').toString()));

--- a/packages/templates/typescript-react-apollo/src/components.handlebars
+++ b/packages/templates/typescript-react-apollo/src/components.handlebars
@@ -5,9 +5,10 @@ import * as React from 'react';
 import gql from 'graphql-tag';
 {{/unless}}
 
-
+{{ blockCommentIf 'Fragments' fragments }}
 {{{generateFragments fragments}}}
 
+{{ blockCommentIf 'Components' operations }}
 {{#each operations }}
     {{#unless @root.config.noNamespaces}}
 export namespace {{toPascalCase name}} {

--- a/packages/templates/typescript-react-apollo/src/helpers/generate-fragments.ts
+++ b/packages/templates/typescript-react-apollo/src/helpers/generate-fragments.ts
@@ -9,7 +9,7 @@ export function generateFragments(fragments: Fragment[], options: Handlebars.Hel
   if (!fragments) {
     return '';
   }
-  const graph = new DepGraph({ circular: true });
+  const graph = new DepGraph<Fragment>({ circular: true });
   fragments.forEach(fragment => {
     graph.addNode(fragment.name, fragment);
   });
@@ -36,12 +36,12 @@ export function generateFragments(fragments: Fragment[], options: Handlebars.Hel
       // Because all of them will have same namespace FooBar
       if (config.noNamespaces) {
         return `
-          export const ${pascalCasedFragmentName}Document = ${gql(fragment, options)};
+          export const ${pascalCasedFragmentName}FragmentDoc = ${gql(fragment, options)};
         `;
       } else {
         return `
           export namespace ${pascalCasedFragmentName} {
-            export const Document = ${gql(fragment, options)};
+            export const FragmentDoc = ${gql(fragment, options)};
           }
         `;
       }

--- a/packages/templates/typescript-react-apollo/src/helpers/to-fragment-name.ts
+++ b/packages/templates/typescript-react-apollo/src/helpers/to-fragment-name.ts
@@ -3,8 +3,8 @@ import { pascalCase } from 'change-case';
 export function toFragmentName(fragmentName: string, options: Handlebars.HelperOptions): string {
   const config = options.data.root.config || {};
   if (config.noNamespaces) {
-    return pascalCase(`${fragmentName}Document`);
+    return pascalCase(`${fragmentName}FragmentDoc`);
   } else {
-    return pascalCase(`${fragmentName}`) + '.Document';
+    return pascalCase(fragmentName) + '.FragmentDoc';
   }
 }

--- a/packages/templates/typescript-react-apollo/tests/components.spec.ts
+++ b/packages/templates/typescript-react-apollo/tests/components.spec.ts
@@ -1,6 +1,7 @@
 import 'graphql-codegen-core/dist/testing';
 import { gql, introspectionToGraphQLSchema, schemaToTemplateContext, transformDocument } from 'graphql-codegen-core';
 import { compileTemplate } from 'graphql-codegen-compiler';
+import { makeExecutableSchema } from 'graphql-tools';
 import config from '../dist';
 import * as fs from 'fs';
 
@@ -193,7 +194,7 @@ describe('Components', () => {
 
     expect(content).toBeSimilarStringTo(`
       export namespace FeedWithRepository {
-        export const Document = gql\`
+        export const FragmentDoc = gql\`
           fragment FeedWithRepository on FeedType {
             id
             commentCount
@@ -202,14 +203,14 @@ describe('Components', () => {
             }
           }
 
-          \${RepositoryWithOwner.Document}
+          \${RepositoryWithOwner.FragmentDoc}
 
         \`;
       }
       `);
     expect(content).toBeSimilarStringTo(`
       export namespace RepositoryWithOwner {
-        export const Document = gql\`
+        export const FragmentDoc = gql\`
           fragment RepositoryWithOwner on Repository {
             full_name
             html_url
@@ -267,7 +268,7 @@ describe('Components', () => {
             }
           }
 
-          \${FeedWithRepository.Document}
+          \${FeedWithRepository.FragmentDoc}
         \`;
       `);
   });
@@ -304,11 +305,11 @@ describe('Components', () => {
             ...SimpleFeed
           }
         }
-        \${SimpleFeed.Document}
+        \${SimpleFeed.FragmentDoc}
       \`
     `);
     expect(content).toBeSimilarStringTo(`
-      const Document = gql\` fragment SimpleFeed on FeedType {
+      const FragmentDoc = gql\` fragment SimpleFeed on FeedType {
         id
         commentCount
       }
@@ -342,5 +343,82 @@ describe('Components', () => {
     const feedWithRepositoryPos = content.indexOf('fragment FeedWithRepository');
     const repositoryWithOwnerPos = content.indexOf('fragment RepositoryWithOwner');
     expect(repositoryWithOwnerPos).toBeLessThan(feedWithRepositoryPos);
+  });
+
+  it('Issue 702 - handle duplicated documents when fragment and query have the same name', async () => {
+    const schema = makeExecutableSchema({
+      typeDefs: `
+        type Event {
+          type: String!
+          name: String!
+        }
+
+        type Query {
+          events: [Event]
+        }
+
+        schema {
+          query: Query
+        }
+      `
+    });
+    const context = schemaToTemplateContext(schema);
+    const documents = gql`
+      fragment event on Event {
+        name
+      }
+
+      query event {
+        events {
+          ...event
+        }
+      }
+    `;
+    const transformedDocument = transformDocument(schema, documents);
+    const compiled = await compileTemplate(
+      {
+        ...config,
+        config: {
+          noNamespaces: false
+        }
+      },
+      context,
+      [transformedDocument],
+      { generateSchema: false }
+    );
+    const content = compiled[0].content;
+
+    expect(content).toBeSimilarStringTo(`
+      export namespace Event {
+        export const FragmentDoc = gql\`
+          fragment event on Event {
+            name
+          }
+        \`;
+      }
+    `);
+
+    expect(content).toBeSimilarStringTo(`
+      export namespace Event {
+        export const FragmentDoc = gql\`
+          fragment event on Event {
+            name
+          }
+        \`;
+      }
+    `);
+
+    expect(content).toBeSimilarStringTo(`
+      export namespace Event {
+        export const Document = gql\`
+          query event {
+            events {
+              ...event
+            }
+          }
+
+          \${Event.FragmentDoc}
+      \`;
+    `);
   });
 });

